go/caddy/pfpki/cloud/intune.go

```diff
@@ -107,35 +109,21 @@ func (cl *Intune) NewCloud(ctx context.Context, name string) error {
 		}
 	}
 
-	oauthConfig, err := adal.NewOAuthConfig(activeDirectoryEndpoint, cl.TenantID)
-
-	// Intune token
-
-	spt, err := adal.NewServicePrincipalToken(*oauthConfig, cl.ClientID, cl.ClientSecret, graphResourceUrl)
-
-	err = spt.Refresh()
-
-	var token adal.Token
-
+	cred, err := azidentity.NewClientSecretCredential(cl.TenantID, cl.ClientID, cl.ClientSecret, nil)
+	// Fetch the token for Graph api
+	tk, err := cred.GetToken(
+		context.TODO(), policy.TokenRequestOptions{Scopes: []string{msGraphResourceUrl + ".default"}},
+	)
 	if err == nil {
-		token = spt.Token()
-		cl.AccessToken = "Bearer " + token.AccessToken
+		cl.AccessToken = "Bearer " + tk.Token
+	} else {
+		log.Print(err)
```

Comment:
Should we return this error?

Reply:
Yes, for troubleshooting purpose it´s really usefull.